### ISSUE TITLE
`bool` is not considered disjoint from NewTypes of `int`, but `Literal[True]` is.

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#4187]
### Summary

Codex spotted this as an intermittent property-test failure while debugging https://github.com/astral-sh/ty/issues/4186. Codex analysis, lightly edited:

---

An `int`-based `NewType` and `bool` should be disjoint: no type can simultaneously be a subtype of both, and thus no variable could ever be considered by a type checker to simultaneously inhabit both types. However, ty considers them overlapping, even though it correctly considers the NewType disjoint from both boolean literals.

```py
from typing import Literal, NewType

from ty_extensions import Intersection, static_assert
from ty_extensions._internal import is_disjoint_from

UserId = NewType("UserId", int)

static_assert(is_disjoint_from(UserId, Literal[True]))   # passes
static_assert(is_disjoint_from(UserId, Literal[False]))  # passes
static_assert(is_disjoint_from(UserId, bool))            # unexpectedly fails

def f(value: Intersection[bool, UserId]) -> None:
    reveal_type(value)  # revealed: bool & UserId; expected: Never
```

This intermittently fails the `structural_negation_subtyping_matches_materialized_negation` property test: structural negation checks the two boolean literals individually, while materialized negation simplifies their union to `bool`, producing inconsistent answers.

The likely cause is the `NewTypeInstance` fallback in `crates/ty_python_semantic/src/types/relation.rs`, which discards the NewType brand and checks its concrete base type against the other type. Consequently, checking `UserId` against `bool` becomes checking `int` against `bool`.

**Comment by carljm** [comment]
See https://github.com/astral-sh/ruff/pull/27522#issuecomment-5196814041 -- I think the correct resolution of this inconsistency may be in the opposite direction: neither `bool` nor `Literal[True]` should be considered disjoint with a NewType of `int`.
